### PR TITLE
Adds User-Agent header.

### LINKED ISSUE #13: Pass a User-Agent header into registry calls.
From https://github.com/GoogleCloudPlatform/minikube-build-tools-for-java/issues/86.

Useful for metrics with registries like GCR.

### PR DESCRIPTION
**Pull request by coollog**
Requires #5 

Fixes #13 

We add this `User-Agent` header to registry calls in order to have registries know the image is built by Jib.

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java [hunk 1]
```diff
@@ -44,6 +45,29 @@
 
   private static final String PROTOCOL = "https";
 
+  @Nullable private static String userAgentSuffix = "";
+
+  // TODO: Inject via a RegistryClientFactory.
```
Comment (coollog) [inline jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java]: Filed https://github.com/google/jib/issues/33

File: jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MojoExceptionBuilder.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.maven;
+
+import org.apache.maven.plugin.MojoExecutionException;
+
+/** Builds a {@link MojoExecutionException} with a useful suggestion. */
+class MojoExceptionBuilder {
```
Comment (elharo) [inline jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MojoExceptionBuilder.java]: This seems over-engineered. Perhaps just use a MojoException?

Comment (elharo) [reply jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MojoExceptionBuilder.java]: Also not used in this PR.

Comment (coollog) [reply jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MojoExceptionBuilder.java]: Oh this was removed in a prior PR - I must have somehow merged incorrectly and caused this to come back. Removed.

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java [hunk 3]
```diff
@@ -45,6 +45,29 @@
...
+  /** Sets a suffix to append to {@code User-Agent} headers. */
+  public static void setUserAgentSuffix(@Nullable String userAgentSuffix) {
+    RegistryClient.userAgentSuffix = userAgentSuffix;
+  }
+
+  // TODO: Add more descriptiveness.
```
Comment (elharo) [inline jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java]: What do you mean? The user-agent shouldn't be too long.

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java [hunk 4]
```diff
@@ -45,6 +45,29 @@
...
+  /** Gets the {@code User-Agent} header to send. */
+  private static String getUserAgent() {
+    String version = RegistryClient.class.getPackage().getImplementationVersion();
+    StringBuilder userAgentBuilder = new StringBuilder();
+    userAgentBuilder.append("jib");
+    if (version != null) {
+      userAgentBuilder.append(" ").append(version);
+    }
+    if (userAgentSuffix != null) {
```
Comment (elharo) [inline jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java]: so you will append an empty string?

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java (hunk 3 above)
Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java]: Not sure yet, removing this for now

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java (hunk 4 above)
Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java]: Ah, `userAgentSuffix` should have been initialized as `null` above. Fixed.

Comment (loosebazooka) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java]: Perhaps a test to cover this?

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java]: Added test.

**Pull request merged**